toolbox/relnotes/main.go

```diff
@@ -644,15 +644,28 @@ func patchRelease(f *os.File, info *ReleaseInfo) {
 }
 
 // extractReleaseNoteFromPR tries to fetch release note from PR body, otherwise uses PR title.
-func extractReleaseNoteFromPR(pr *github.Issue) string {
+func extractReleaseNoteFromPR(pr *github.Issue, actionRequired bool) string {
+	var releaseNote string
 	// Regexp Example:
 	// This regexp matches the release note section in Kubernetes pull request template:
 	// https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-	re, _ := regexp.Compile("```release-note\r\n(.+)\r\n```")
+	re := regexp.MustCompile("(?s)```release-note\r\n(.+)\r\n```")
 	if note := re.FindStringSubmatch(*pr.Body); note != nil {
-		return note[1]
+		releaseNote = strings.TrimSpace(note[1])
+	} else {
+		// Use PR title if unable to find release note
+		releaseNote = strings.TrimSpace(*pr.Title)
+	}
+	// We do this for action required PRs only, to avoid removing "action required" message from
+	// some PR that accidentally gets marked as non-action-required PR.
```

Comment:
If `actionRequired` is true here, it means the PR is already in `releaseActionRequiredPRs` and marked as an action-required PR, so I feel like the comment is a bit confusing.

Reply:
I'm thinking about PRs that might be manually labelled as `release-note` (non-action-required), but has "action required" text in their release note. In this case, we should not remove the "action required" text, to allow maintainers/users to notice the inconsistency and update the label or release note as needed. (the possibility for this case to happen is very low I suppose)

The comment is more about "why we don't remove text if `actionRequired` is false". I will update the comment.

Reply:
Makes sense to me. Thanks for clarifying!